infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespace.java

```diff
@@ -227,6 +298,19 @@ private void create(String namespaceName, KubernetesClient client)
     }
   }
 
+  private void update(Namespace namespace, KubernetesClient client) throws InfrastructureException {
+    try {
+      client.namespaces().createOrReplace(namespace);
```

Comment:
Is `createOrReplace` the correct logic here, or would an exception when namespace doesn't already exist be more appropriate?

Reply:
This is a private method called from a place where we are sure the namespace already exists and we only update it with new metadata.

There is no "update" on the kubernetes API for namespaces as far as I can tell. Hence the choice of this call.

Reply:
Fair enough, I assumed there was a `#replace` or `#update` method available.